src/legacy/ui/public/courier/search_source/search_source.js

```diff
@@ -173,38 +141,6 @@ export function SearchSourceProvider(Promise, Private, config) {
         throw new Error(`Can't set field '${field}' on SearchSource. Acceptable fields are: ${FIELDS.join(', ')}.`);
       }
 
-      if (field === 'index') {
```

Comment:
This logic was over-complicated and moved [here](https://github.com/elastic/kibana/pull/34508/files#diff-181c01982cb42305fd5488a4f5d84befR389)